web/web.py

```diff
@@ -36,67 +38,99 @@ def __init__(self, bot):
-            except ReportException:  # report not found
-                issue_labels = [lab['name'] for lab in issue['labels']]
-                if EXEMPT_LABEL in issue_labels:
-                    return
-
-                report = Report.new_from_issue(repo_name, issue)
-                if not issue['title'].startswith(report.report_id):
-                    formatted_title = f"{report.report_id} {report.title}"
-                    await GitHubClient.get_instance().rename_issue(repo_name, issue['number'], formatted_title)
-
-                # await GitHubClient.get_instance().add_issue_to_project(report.github_issue, report.is_bug)
-                await GitHubClient.get_instance().add_issue_comment(repo_name, issue['number'],
-                                                                    f"Tracked as `{report.report_id}`.")
-                await report.update_labels()
-
-            await report.unresolve(ContextProxy(self.bot), open_github_issue=False)
-            report.commit()
+            await self.report_opened(data)
         elif action in ("labeled", "unlabeled"):
-            try:
-                report = Report.from_github(repo_name, issue_num)
-            except ReportException:  # report not found
-                return  # oh well
+            await self.report_labeled(data)
 
-            if len([l for l in issue['labels'] if any(n in l['name'] for n in PRI_LABEL_NAMES)]) > 1:
-                return  # multiple priority labels
+    async def report_closed(self, data):
+        issue = data['issue']
+        issue_num = issue['number']
+        repo_name = data['repository']['full_name']
+        try:
+            report = Report.from_github(repo_name, issue_num)
+        except ReportException:  # report not found
+            return  # oh well
+
+        pend = data['sender']['login'] == constants.OWNER_GITHUB
 
-            label_names = [l['name'] for l in issue['labels']]
+        await report.resolve(ContextProxy(self.bot), close_github_issue=False, pend=pend)
+        report.commit()
+
+    async def report_opened(self, data):
+        issue = data['issue']
+        issue_num = issue['number']
+        repo_name = data['repository']['full_name']
+        # is the issue new?
+        try:
+            report = Report.from_github(repo_name, issue_num)
+        except ReportException:  # report not found
+            issue_labels = [lab['name'] for lab in issue['labels']]
+            if EXEMPT_LABEL in issue_labels:
+                return None
+
+            report = Report.new_from_issue(repo_name, issue)
+            if not issue['title'].startswith(report.report_id):
+                formatted_title = f"{report.report_id} {report.title}"
+                await GitHubClient.get_instance().rename_issue(repo_name, issue['number'], formatted_title)
+
+            # await GitHubClient.get_instance().add_issue_to_project(report.github_issue, report.is_bug)
```

Comment:
Is this commented out for a reason?

Reply:
We chatted about it earlier iirc - the projects API for GitHub is somewhat immature, but I did write the code that would *theoretically* work once it is more stable, so I just commented it out instead of deleting it